src/set_theory/cardinal/basic.lean

```diff
@@ -564,22 +565,27 @@ cardinal.wf.conditionally_complete_linear_order_with_bot 0 $ le_antisymm (cardin
 
 instance wo : @is_well_order cardinal.{u} (<) := ⟨cardinal.wf⟩
 
+/-- The set in the definition of `cardinal.succ` is nonempty. -/
+theorem succ_nonempty (c : cardinal) : {c' : cardinal | c < c'}.nonempty := ⟨_, cantor c⟩
+
 /-- The successor cardinal - the smallest cardinal greater than
   `c`. This is not the same as `c + 1` except in the case of finite `c`. -/
 def succ (c : cardinal) : cardinal :=
 Inf {c' | c < c'}
 
-theorem succ_nonempty (c : cardinal) : {c' : cardinal | c < c'}.nonempty :=
-⟨_, cantor _⟩
-
-theorem lt_succ_self (c : cardinal) : c < succ c :=
+theorem lt_succ (c : cardinal) : c < succ c :=
```

Comment:
If you swap things around can `order.lt_succ` prove this?

Reply:
Again, this is directly used to build the instance itself, so this doesn't quite work.